Make roles and responsibilities its own page (#22001)
kubernetes/website: content/en/docs/contribute/participate/_index.md

within the Kubernetes project, focused on writing, updating, and maintaining
the documentation for Kubernetes as a whole. See
[SIG Docs from the community github repo](https://github.com/kubernetes/community/tree/master/sig-docs)
for more information about the SIG.

SIG Docs welcomes content and reviews from all contributors. Anyone can open a
pull request (PR), and anyone is welcome to file issues about content or comment
on pull requests in progress.

You can also become a [member](/docs/contribute/participating/roles-and-responsibilities/#members),
[reviewer](/docs/contribute/participating/roles-and-responsibilities/#reviewers), or [approver](/docs/contribute/participating/roles-and-responsibilities/#approvers). These roles require greater
access and entail certain responsibilities for approving and committing changes.
See [community-membership](https://github.com/kubernetes/community/blob/master/community-membership.md)
for more information on how membership works within the Kubernetes community.

The rest of this document outlines some unique ways these roles function within
SIG Docs, which is responsible for maintaining one of the most public-facing

> aspects of Kubernetes -- the Kubernetes website and documentation.

Reviewer: Excellent work. 👍

<!-- body -->

> ## SIG Docs chairperson

Reviewer: nit: s/chairperson/chair throughout

```suggestion
## SIG Docs chair
```

Each SIG, including SIG Docs, selects one or more SIG members to act as

> chairpersons. These are points of contact between SIG Docs and other parts of

Reviewer: ```suggestion
chairs. Chairs are points of contact between SIG Docs and other parts of
```

the Kubernetes organization. They require extensive knowledge of the structure
of the Kubernetes project as a whole and how SIG Docs works within it. See
[Leadership](https://github.com/kubernetes/community/tree/master/sig-docs#leadership)

> for the current list of chairpersons.

Reviewer: ```suggestion
for the current list of chairs.
```

## SIG Docs teams and automation

> Automation in SIG Docs relies on two different mechanisms for automation:

Reviewer 2: Is it possible to use one instance of the word `automation`:
To automate the website processes, SIG Docs relies on GitHub groups and OWNERS files.
OR
Automation in SIG Docs relies on GitHub groups and OWNERS files.

Reviewer: @kbhawkey Good suggestion!

GitHub groups and OWNERS files.

### GitHub groups

The SIG Docs group defines two teams on GitHub:

 - [@kubernetes/sig-docs-maintainers](https://github.com/orgs/kubernetes/teams/sig-docs-maintainers)
 - [@kubernetes/sig-docs-pr-reviews](https://github.com/orgs/kubernetes/teams/sig-docs-pr-reviews)

Each can be referenced with their `@name` in GitHub comments to communicate with

> everyone in that group.

Reviewer: This whole section is ✨

> These teams overlap, but do not exactly match, the groups used by the automation
> tooling. For assignment of issues, pull requests, and to support PR approvals,

Reviewer: After more thought:
```suggestion
Prow and GitHub times sometimes overlap without matching exactly. For assignment of issues, pull requests, and to support PR approvals,
```

the automation uses information from OWNERS files.

> ### OWNERS files and front-matter

Reviewer: ```suggestion
### OWNERS files and front matter
```

> The Kubernetes project uses an automation tool called prow for automation

Reviewer: ```suggestion
The Kubernetes project uses a tool called Prow for automation
```

related to GitHub issues and pull requests. The
[Kubernetes website repository](https://github.com/kubernetes/website) uses

> two [prow plugins](https://github.com/kubernetes/test-infra/tree/master/prow/plugins):

Reviewer: ```suggestion
three [prow plugins](https://github.com/kubernetes/test-infra/tree/master/prow/plugins):
```

- blunderbuss

> - approve

Reviewer: `Website` lets us use the `/website` command.

```suggestion
- approve
- website
```

Reviewer 3: Looking at the prow plugins, I think this entire section will need to be revised.

The actual `k/website` enabled ones are (https://github.com/kubernetes/test-infra/blob/master/config/prow/plugins.yaml#L725):
```
  kubernetes/website:
  - milestone
  - project
```

`k/*` enables:
```
  kubernetes:
  - approve
  - assign
  - blunderbuss
  - cat
  - cla
  - dog
  - golint
  - goose
  - heart
  - help
  - hold
  - invalidcommitmsg
  - label
  - lgtm
  - lifecycle
  - owners-label
  - pony
  - retitle
  - shrug
  - sigmention
  - size
  - skip
  - slackevents
  - trigger
  - verify-owners
  - welcome
  - wip
  - yuks
```

Maybe there's a different way to get the point across?

/honk
/woof
/pony

Reviewer 4: Telling readers about at least the top 5 or 6 commands is going to be really useful.

> These two plugins use the

Reviewer: ![onetwofive](https://user-images.githubusercontent.com/3210446/85645124-abd7b900-b64d-11ea-8aac-c82033524008.gif)

```suggestion
These three plugins use the
```

[OWNERS](https://github.com/kubernetes/website/blob/master/OWNERS) and
[OWNERS_ALIASES](https://github.com/kubernetes/website/blob/master/OWNERS_ALIASES)
files in the top level of the `kubernetes/website` GitHub repository to control

> how prow works within the repository.

Reviewer: ```suggestion
how Prow works within the repository.
```

> An OWNERS file contains a list of people who are SIG Docs reviewers and

Reviewer: ```suggestion
An OWNERS file contains a list of SIG Docs reviewers and
```

approvers. OWNERS files can also exist in subdirectories, and can override who

> can act as a reviewer or approver of files in that subdirectory and its

Reviewer: ```suggestion
acts as a reviewer or approver of files in that subdirectory and its
```

> descendents. For more information about OWNERS files in general, see

Reviewer 2: nit: descendants?

[OWNERS](https://github.com/kubernetes/community/blob/master/contributors/guide/owners.md).

In addition, an individual Markdown file can list reviewers and approvers in its

> front-matter, either by listing individual GitHub usernames or GitHub groups.

Reviewer: ```suggestion
front matter, either by listing individual GitHub usernames or GitHub groups.
```

> The combination of OWNERS files and front-matter in Markdown files determines

Reviewer: ```suggestion
The combination of OWNERS files and front matter in Markdown files determines
```

Author: I'm enjoying how nit-picky these changes are ✨

the advice PR owners get from automated systems about who to ask for technical
and editorial review of their PR.

## How merging works

> When a pull request is merged to the branch used to publish content (currently

Reviewer: Active voice
```suggestion
When a pull request merges to the branch used to publish content (currently
```

> `master`), that content is published to http://kubernetes.io. To ensure that

Reviewer 2: nit: Could leave out `(currently master)`.

> the quality of our published content is high, we limit merging pull requests to

Reviewer 2: nit: `available to the world`
Docs are published publicly. The changes are immediately available (to the site).

> SIG Docs approvers. Here's how it works.

Reviewer: 👍

> - When a pull request has both the `lgtm` and `approve` labels, has no `hold`

Reviewer 3: ```suggestion
- When a pull request has both the `lgtm` and `approve` labels, has no `hold` or `work-in-progress`
```

  labels, and all tests are passing, the pull request merges automatically.
- Kubernetes organization members and SIG Docs approvers can add comments to
  prevent automatic merging of a given pull request (by adding a `/hold` comment
  or withholding a `/lgtm` comment).
- Any Kubernetes member can add the `lgtm` label by adding a `/lgtm` comment.
- Only SIG Docs approvers can merge a pull request

>   by adding an `/approve` comment. Some approvers also perform additional

Reviewer 3: ```suggestion
  by adding an `/approve` comment. 
  
  Some approvers also perform additional
```
